File: src/0026-Remove-Duplicates-from-Sorted-Array/remove_duplicates_from_sorted_array.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int removeDuplicates(std::vector<int>& nums) {
        int removing = nums[nums.size() - 1];
        int k = nums.size();
        int nums_len = nums.size() - 2;

        while (nums_len >= 0) {
            if (nums[nums_len] == removing) {
                nums.erase(nums.begin() + nums_len);
                k--;
            }
            else {
                removing = nums[nums_len];
            }
            nums_len--;
        }
        return k;
    }
};

int main() {
    std::vector<int> nums = {0,0,1,1,1,2,2,3,3,4};
    
    Solution solution;

    std::cout << solution.removeDuplicates(nums) << " ";
    for (int i : nums) {
        std::cout << i << " ";
    }

    return 0;
}

File: src/0026-Remove-Duplicates-from-Sorted-Array/remove_duplicates_from_sorted_array.py
from typing import List

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        removing: int = nums[-1]
        k: int = len(nums)
        nums_len: int  = len(nums) - 2

        while nums_len >= 0:
            if nums[nums_len] == removing:
                nums.pop(nums_len)
                k -= 1
            else:
                removing = nums[nums_len]
            nums_len -= 1
        return k

if __name__ == "__main__":
    nums: List[int] = [0,0,1,1,1,2,2,3,3,4]
    
    solution = Solution()
    
    print(f"{solution.removeDuplicates(nums)} {nums}")